src/main/java/org/assertj/core/extention/SoftlyExtension.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.extention;
+
+import org.assertj.core.annotations.Softly;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.internal.Failures;
+import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
+import org.junit.jupiter.api.extension.BeforeTestExecutionCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.TestInstancePostProcessor;
+import org.junit.platform.commons.support.AnnotationSupport;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+
+/**
+ * <p>Injects an instance of {@link SoftAssertions} to any Field of type
+ * {@code SoftAssertions} that has the annotation {@link Softly}
+ * before each test method execution.
+ * {@link SoftAssertions#assertAll()} executes after each test
+ * method only when {@link Softly#assertAfter()} return {@code true}
+ * By default {@code assertAfter()} is {@code true} to make sure
+ * global verification still happens even when {@code softly.assertAll()}
+ * is not called
+ *</p>
+ *
+ * <p>
+ * Inspired by Mockito @Captor annotation.
+ * </p>
+ * @author Arthur Mita
+ **/
+public class SoftlyExtension implements TestInstancePostProcessor, BeforeTestExecutionCallback, AfterTestExecutionCallback {
+
+  private static final Logger LOGGER = Logger.getLogger(SoftlyExtension.class.getName());
+  private static final Namespace ASSERT_J = Namespace.create("org.assertj.core");
+  private static final String TEST_INSTANCE = "testInstance";
+  private final List<Field> fields = new ArrayList<>();
```

Comment:
I think it should be only one field, not sure why we would need several

Reply:
A lot of the inspiration came from Mockito's `@Captor` which allows one to have more that one `ArgumentCapture` field but then `ArgumentCapture` accepts generics, so it makes sense to have more than one instance. I will remove it.